client/ayon_core/hosts/resolve/plugins/load/load_media.py

```diff
@@ -0,0 +1,363 @@
+    if not usage:
+        return []
+
+    if project is None:
+        project = lib.get_current_project()
+
+    matching_items = []
+    unique_id = media_pool_item.GetUniqueId()
+    for timeline_idx in range(project.GetTimelineCount()):
+        timeline = project.GetTimelineByIndex(timeline_idx + 1)
+
+        # Consider audio and video tracks
+        for track_type in ["video", "audio"]:
+            for track_idx in range(timeline.GetTrackCount(track_type)):
+                timeline_items = timeline.GetItemListInTrack(track_type,
+                                                             track_idx + 1)
+                for timeline_item in timeline_items:
+                    timeline_item_mpi = timeline_item.GetMediaPoolItem()
+                    if not timeline_item_mpi:
+                        continue
+
+                    if timeline_item_mpi.GetUniqueId() == unique_id:
+                        matching_items.append((timeline, timeline_item))
+                        usage -= 1
+                        if usage <= 0:
+                            # If there should be no usage left after this found
+                            # entry we return early
+                            return matching_items
+
+    return matching_items
+
+
+class LoadMedia(LoaderPlugin):
+    """Load a subset as media pool item."""
+
+    families = ["render2d", "source", "plate", "render", "review"]
+
+    representations = ["*"]
+    extensions = set(
+        ext.lstrip(".") for ext in IMAGE_EXTENSIONS.union(VIDEO_EXTENSIONS)
+    )
+
+    label = "Load media"
+    order = -20
+    icon = "code-fork"
+    color = "orange"
+
+    options = [
+        BoolDef(
+            "load_to_timeline",
+            label="Load to timeline",
+            default=True,
+            tooltip="Whether on load to automatically add it to the current "
+                    "timeline"
+        ),
+        BoolDef(
+            "load_once",
+            label="Re-use existing",
+            default=True,
+            tooltip="When enabled - if this particular version is already"
+                    "loaded it will not be loaded again but will be re-used."
+        )
+    ]
+
+    # for loader multiselection
+    timeline = None
+
+    # presets
+    clip_color_last = "Olive"
+    clip_color = "Orange"
+
+    bin_path = "Loader/{representation[context][hierarchy]}/{asset[name]}"
+
+    metadata = [
+        {
+            "name": "Comments",
+            "value": "{version[data][comment]}"
+        },
+        {
+            "name": "Shot",
+            "value": "{asset[name]}"
+        },
+        {
+            "name": "Take",
+            "value": "{subset[name]} v{version[name]:03d}"
+        },
+        {
+            "name": "Clip Name",
+            "value": (
+                "{asset[name]} {subset[name]} "
+                "v{version[name]:03d} ({representation[name]})"
+            )
+        }
+    ]
```

Comment:
Why is this suddenly a list instead of a dict? This seems less efficient and less sensible since we shouldn't ever need duplicate keys either. Can we keep this a `dict`?

Reply:
This is just a prep for settings in AYON. There is not the same editable dictionary objects as in OpenPype.

Reply:
Oh boy, that's awful design really. I understand that maybe the backend is set up to be ordered for nice front-end but I'd expect the settings Pydantic stuff could auto-convert to a Python `dict` just fine, no?

It's really extra confusing here already that it's `name` instead of `key` (and should admittedly be just a dict)

Reply:
This is marked resolved but I believe hasn't changed in data structure itself. Is there anything we can do to actually improve this becoming an actual `dict` datatype as opposed to the odd list structure for what is essentially intended to be a dict on the python side.

It's _likely_ slower, uses more memory but above all it's harder to read and an unnecessary bloated data structure for something that is essentially a `dict`.